src/Validation/ScannableURLProvider.php

```diff
@@ -69,9 +70,11 @@ public function get_urls( $offset = 0 ) {
 			foreach ( $public_post_types as $post_type ) {
 				$post_ids = $this->get_posts_that_support_amp( $this->get_posts_by_type( $post_type, $i, 1 ) );
 				if ( ! empty( $post_ids[0] ) ) {
-					$urls[] = [
-						'url'  => get_permalink( $post_ids[0] ),
-						'type' => $post_type,
+					$post_type_object = get_post_type_object( $post_type );
+					$urls[]           = [
+						'url'   => get_permalink( $post_ids[0] ),
+						'type'  => $post_type,
+						'label' => empty( $post_type_object->labels->name ) ? $post_type : $post_type_object->labels->name,
```

Comment:
Following up on what you said in https://github.com/ampproject/amp-wp/pull/6596#discussion_r717788251:

> Note that we may still have issues with the custom taxonomy and the custom post type labels. For instance, WooCommerce registers a `product_cat` taxonomy. Its `name` is "Product categories" and its `singular_name` is just "Category". The core Category `name` is "Categories" and the `singular_name` is "Category", too.
>
> I went with the `name` label since it best differentiates the taxonomies. The drawback is that it is usually a plural noun and probably should be used standalone (unless we're okay with having "Categories Archive" or "Tags Archive").

I think that WooCommerce may be in error here to not have chosen “Product category” as the singular name. I see they have also used “Categories” as the `label` too: https://github.com/woocommerce/woocommerce/blob/55ae972e8abf5ef5a53bbf6338f433f8927bff84/includes/class-wc-post-types.php#L93-L109

Therefore, I think we can just go ahead and use the `singular_name`.

```suggestion
						'label' => $post_type_object->labels->singular_name ?: $post_type,
```